crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -234,6 +391,28 @@ impl<F: Clone> PostUpdateTracker<F, PaymentData<F>, types::PaymentsSyncData> for
         ))
         .await
     }
+
+    async fn save_pm_and_mandate<'b>(
+        &self,
+        state: &AppState,
+        resp: &types::RouterData<F, types::PaymentsSyncData, types::PaymentsResponseData>,
+        merchant_account: &domain::MerchantAccount,
+        _key_store: &domain::MerchantKeyStore,
+        payment_data: &mut PaymentData<F>,
+    ) -> CustomResult<(), errors::ApiErrorResponse>
+    where
+        F: 'b + Clone + Send + Sync,
+    {
+        update_payment_method_status_and_ntid(
```

Comment:
we cannot make `customer_id` mandatory here, if the connector call fails, we should not be doing this check at all.